actionpack/test/controller/live_stream_test.rb

```diff
@@ -388,8 +388,14 @@ def test_exception_handling_plain_text
     end
 
     def test_exception_callback_when_committed
+      current_threads = Thread.list
+
       capture_log_output do |output|
         get :exception_with_callback, format: 'text/event-stream'
+
+        # Wait on the execution of all threads
+        (Thread.list - current_threads).each { |t| t.join }
```

Comment:
I narrowed it down to https://github.com/rails/rails/blob/2ad9afe4ff2c12d1a9e4a00d9009d040e636c9b0/actionpack/lib/action_controller/metal/live.rb#L246-L273. If the assertion runs before the second threads complete, the test will fail.